src/schematic/traits.rs

```diff
@@ -29,10 +29,10 @@ pub const INGRESS: &str = "ingress";
 pub const AUTOSCALER: &str = "auto-scaler";
 pub const MANUAL_SCALER: &str = "manual-scaler";
 pub const VOLUME_MOUNTER: &str = "volume-mounter";
-pub const INGRESS_V1ALPHA1: &str = "ingress.core.oam.dev/v1alpha1";
-pub const AUTOSCALER_V1ALPHA1: &str = "auto-scaler.core.oam.dev/v1alpha1";
-pub const MANUAL_SCALER_V1ALPHA1: &str = "manual-scaler.core.oam.dev/v1alpha1";
-pub const VOLUME_MOUNTER_V1ALPHA1: &str = "volume-mounter.core.oam.dev/v1alpha1";
+pub const INGRESS_V1ALPHA1: &str = "ingress";
```

Comment:
>INGRESS_V1ALPHA1 [](start = 10, length = 16)

You have to make change to trait_manager.rs - load_traits() method to remove the case statements  "INGRESS", "AUTOSCALER" and others. Additionally the trait object has from_param() method that needs to be removed as well.

Reply:
Why do I need to change that?
The current behavior with the change looks correct to me now.

Reply:
If you look at the switch stmt, "INGRESS_V1ALPHA1" is before "INGRESS" case that is why it is working. If you change the ordering it will break. With this change, we have two case options with same value and because of case statement ordering it works in your case.

Earlier params(key-value pair) was used as the way to get input and recent change uses fixed schema (via properties). Since we are moving to fixed schema old remnants that is using params should be removed.


---
In reply to: [367710920](https://github.com/oam-dev/rudr/pull/516#discussion_r367710920) [](ancestors = 367710920)

Reply:
Hi @iyyappam 

Thanks for your explanations. I got your meanings.

Moreover, would you mind tell me your Slack ID? That would be easier for me to ask you additional questions and discuss issues :)

Reply:
> has from_param() method which are not needed anymore.

done.